core/lib/spree/preferences/configuration.rb

```diff
@@ -38,18 +38,35 @@ class Configuration
     #     defaults. Set via {#load_defaults}
     attr_reader :loaded_defaults
 
+    attr_reader :load_defaults_called
+
     def initialize
       @loaded_defaults = Spree.solidus_version
+      @load_defaults_called = false
     end
 
     # @param [String] Solidus version from which take defaults when not
     # overriden.
     # @see #load_defaults
     def load_defaults(version)
       @loaded_defaults = version
+      @load_defaults_called = true
       reset
     end
 
+    def check_load_defaults_called(instance_constant_name = nil)
+      return if load_defaults_called || !Spree::Core.solidus_installed?
+
+      target_name = instance_constant_name || "#{self.class.name}.new"
+      Spree::Deprecation.warn <<~MSG
+        Please, indicate wich Solidus version defaults the application must use.
+        The method `#{target_name}.load_defaults` must be called from the Solidus initializer. E.g.:
+
+          config.load_defaults('#{Spree.solidus_version}')
```

Comment:
I think it could be more intuitive if we use `Spree::Config.load_defaults(...)` dynamically based on the module user needs to change (`Api`, `Backend`, etc). Do you think it is possible?

Reply:
Please, take a look at commit 5f81fee. I've introduced a `Spree.load_defaults` method, which forwards to all available components. I prefer to have it on `Spree` instead of `Spree::Config`, as `Spree::Config` is an instance of the `Core` configuration, and we need fine-grained control to allow users to enable new defaults one by one, component by component.

The new method simplifies our test setup to avoid the warning, so thanks for pointing that out!!